test/e2e/auth/pod_security_policy.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/security/apparmor"
+	"k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("PodSecurityPolicy", func() {
+	f := framework.NewDefaultFramework("podsecuritypolicy")
+	f.SkipDefaultPSPBinding = true
+
+	// Client that will impersonate the default service account, in order to run
+	// with reduced privileges.
+	var c clientset.Interface
+	var ns string // Test namespace, for convenience
+	BeforeEach(func() {
+		if !framework.IsPodSecurityPolicyEnabled(f) {
+			framework.Skipf("PodSecurityPolicy not enabled")
+		}
+		ns = f.Namespace.Name
+
+		By("Creating a kubernetes client that impersonates the default service account")
+		config, err := framework.LoadConfig()
+		framework.ExpectNoError(err)
+		config.Impersonate = restclient.ImpersonationConfig{
+			UserName: serviceaccount.MakeUsername(ns, "default"),
+			Groups:   serviceaccount.MakeGroupNames(ns),
+		}
+		c, err = clientset.NewForConfig(config)
+		framework.ExpectNoError(err)
+
+		By("Binding the edit role to the default SA")
+		framework.BindClusterRole(f.ClientSet.RbacV1beta1(), "edit", ns,
+			rbacv1beta1.Subject{Kind: rbacv1beta1.ServiceAccountKind, Namespace: ns, Name: "default"})
+	})
+
+	It("should forbid pod creation when no PSP is available", func() {
+		By("Running a restricted pod")
+		_, err := c.Core().Pods(ns).Create(restrictedPod(f, "restricted"))
+		expectForbidden(err)
+	})
+
+	It("should enforce the restricted PodSecurityPolicy", func() {
+		By("Binding the restricted policy for the test service account")
+		framework.BindClusterRole(f.ClientSet.RbacV1beta1(), "podsecuritypolicy:restricted:v1", ns,
```

Comment:
I'd expect the e2e test to define the PSPs, roles, and bindings, rather than assuming a particular deployment method or addons